polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -733,22 +810,74 @@ impl State {
 		);
 
 		if let Some(approval_entry) = candidate_entry.approval_entry(&block_hash) {
-			let required_tranches = approval_checking::tranches_to_approve(
-				approval_entry,
-				candidate_entry.approvals(),
-				tranche_now,
-				block_tick,
-				no_show_duration,
-				session_info.needed_approvals as _,
-			);
+			let TranchesToApproveResult { required_tranches, total_observed_no_shows } =
+				approval_checking::tranches_to_approve(
+					approval_entry,
+					candidate_entry.approvals(),
+					tranche_now,
+					block_tick,
+					no_show_duration,
+					session_info.needed_approvals as _,
+				);
 
-			let status = ApprovalStatus { required_tranches, block_tick, tranche_now };
+			let status = ApprovalStatus {
+				required_tranches,
+				block_tick,
+				tranche_now,
+				last_no_shows: total_observed_no_shows,
+			};
 
 			Some((approval_entry, status))
 		} else {
 			None
 		}
 	}
+
+	// Returns the approval voting from the RuntimeApi.
+	// To avoid crossing the subsystem boundary every-time we are caching locally the values.
```

Comment:
Forgot to delete the comment, I was having a cache locally, but then the other reviewers suggested to remove it, because it is not needed because we already have a cache and crossing the boundary is not that expensive, so I remove the cache and forgot the comment.